File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys
input = sys.stdin.readline

N = int(input())
arr = {}
for _ in range(N):
    a, b = input().split()

    if b == "enter":
        arr[a] = 1
    elif b == 'leave':
        arr[a] = 0

arr = sorted(arr.items(), reverse=True)
for i in arr:
    if i[1] == 1:
        print(i[0])

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split("\n")

let arr = new Set()
for (let i = 1; i < input.length; i++) {
    const line = input[i].split(" ")
    const name  = line[0].trim()
    const state = line[1].trim()
    if (state === 'enter') {arr.add(name)}
    if (state === 'leave') {arr.delete(name)}
}

answer = [...arr].sort().reverse()
console.log(answer.join("\n"))